include/thread.h

```diff
@@ -28,6 +28,7 @@ typedef struct thread {
   enum { TDS_INACTIVE = 0x0, TDS_WAITING, TDS_READY, TDS_RUNNING } td_state;
   uint32_t td_flags;           /* TDF_* flags */
   volatile uint32_t td_csnest; /* critical section nest level */
+  int td_exitcode;
```

Comment:
We should discuss the meaning of this field. Initial idea is to map kernel exit codes (i.e. `errno` and received signal) to negative numbers, whereas positive numbers are free to use by userspace. What do you think?